doc/index.rst

```diff
@@ -0,0 +1,600 @@
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol on Python 3.11 and lower.
+
+   .. versionadded:: 4.6.0
+
+.. class:: SupportsComplex
+
+   See :py:class:`typing.SupportsComplex`.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol on Python 3.11 and lower.
+
+   .. versionadded:: 4.6.0
+
+.. class:: SupportsFloat
+
+   See :py:class:`typing.SupportsFloat`.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol on Python 3.11 and lower.
+
+   .. versionadded:: 4.6.0
+
+.. class:: SupportsIndex
+
+   See :py:class:`typing.SupportsIndex`. In ``typing`` since 3.8.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol on Python 3.11 and lower.
+
+   .. versionchanged:: 4.6.0
+
+      Backported the performance improvements from Python 3.12.
+
+.. class:: SupportsInt
+
+   See :py:class:`typing.SupportsInt`.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol on Python 3.11 and lower.
+
+   .. versionadded:: 4.6.0
+
+.. class:: SupportsRound
+
+   See :py:class:`typing.SupportsRound`.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol on Python 3.11 and lower.
+
+   .. versionadded:: 4.6.0
+
+Decorators
+~~~~~~~~~~
+
+.. decorator:: dataclass_transform(*, eq_default=False, order_default=False,
+                                   kw_only_default=False, frozen_default=False,
+                                   field_specifiers=(), **kwargs)
+
+   See :py:func:`typing.dataclass_transform`. In ``typing`` since 3.11.
+
+   Python 3.12 adds the ``frozen_default`` parameter; ``typing_extensions``
+   backports this parameter.
+
+   .. versionadded:: 4.1.0
+
+   .. versionchanged:: 4.2.0
+
+      The ``field_descriptors`` parameter was renamed to ``field_specifiers``.
+      For compatibility, the decorator now accepts arbitrary keyword arguments.
+
+   .. versionchanged:: 4.5.0
+
+      The ``frozen_default`` parameter was added.
+
+.. decorator:: deprecated(msg, *, category=DeprecationWarning, stacklevel=1)
+
+   See :pep:`702`. Experimental; not yet part of the standard library.
+
+   .. versionadded:: 4.5.0
+
+.. decorator:: final
+
+   See :py:func:`typing.final`. In ``typing`` since 3.8.
+
+   Since Python 3.11, this decorator supports runtime introspection
+   by setting the ``__final__`` attribute wherever possible; ``typing_extensions.final``
+   backports this feature.
+
+   .. versionchanged:: 4.1.0
+
+      The decorator now attempts to set the ``__final__`` attribute on decorated objects.
+
+.. decorator:: overload
+
+   See :py:func:`typing.overload`.
+
+   Since Python 3.11, this decorator supports runtime introspection
+   through :func:`get_overloads`; ``typing_extensions.overload``
+   backports this feature.
+
+   .. versionchanged:: 4.2.0
+
+      Added to provide support for introspection through :func:`get_overloads` was added.
```

Comment:
```suggestion
      Introspection support via :func:`get_overloads` was added.
```

Reply:
There was no `typing_extensions.overload` before this, so the note is here to explain why we added. I should actually change the line above to "versionadded", not "versionchanged".

Reply:
Are you sure? I can see it as far back as 3.10.1: https://github.com/python/typing_extensions/blob/0c70227e221866b74b5cf6b3e251bc67864781ef/src_py3/typing_extensions.py#L693-L719

Reply:
Oh you're right, I misremembered. I wonder why; according to https://docs.python.org/3.10/library/typing.html#typing.overload it's always been in `typing`.

Reply:
The past is a foreign country: they do things differently there